internal/helm/helm.go

```diff
@@ -82,92 +82,93 @@ func Deploy(logger logr.Logger, parameters ChartParameters) error {
 		return fmt.Errorf("Unable to deploy, chart %s not found", parameters.Chart)
 	}
 
-	// YAML string - TODO ? Use unstructured as intermediary to
-	// marshal yaml from ? Instead of direct generation of a
-	// string ?
-
-	configurationNames := `[]`
-	if len(parameters.Configurations) > 0 {
-		configurationNames = fmt.Sprintf(`["%s"]`, strings.Join(parameters.Configurations, `","`))
+	// Local type definitions for proper marshalling of the
+	// `values.yaml` to hand to helm from the chart parameters.
+
+	type routeParam struct {
+		Id     string `yaml:"id"`
+		Domain string `yaml:"domain"`
+		Path   string `yaml:"path"`
+		Secret string `yaml:"secret,omitempty"`
+	}
+	type epinioParam struct {
+		AppName        string               `yaml:"appName"`
+		Env            []models.EnvVariable `yaml:"env"`
+		ImageUrl       string               `yaml:"imageURL"`
+		Ingress        string               `yaml:"ingress,omitempty"`
+		ReplicaCount   int32                `yaml:"replicaCount"`
+		Routes         []routeParam         `yaml:"routes"`
+		Configurations []string             `yaml:"configurations"`
+		StageID        string               `yaml:"stageID"`
+		TlsIssuer      string               `yaml:"tlsIssuer"`
+		Username       string               `yaml:"username"`
+		Start          string               `yaml:"start,omitempty"`
+	}
+	type chartParam struct {
+		Epinio epinioParam `yaml:"epinio"`
+	}
+
+	// Fill values.yaml structure
+
+	params := chartParam{
+		Epinio: epinioParam{
+			AppName:        parameters.Name,
+			Env:            parameters.Environment.List(),
+			ImageUrl:       parameters.ImageURL,
+			ReplicaCount:   parameters.Instances,
+			Configurations: parameters.Configurations,
+			StageID:        parameters.StageID,
+			TlsIssuer:      viper.GetString("tls-issuer"),
+			Username:       parameters.Username,
+			// Ingress, Start, Routes: see below
+		},
 	}
 
-	environment := `[]`
-	if len(parameters.Environment) > 0 {
-		// TODO: Simplify the chain of conversions. Single `AsYAML` ?
-		environment = fmt.Sprintf(`[ %s ]`, strings.Join(parameters.Environment.List().Assignments(),
-			","))
+	// TODO: Is this properly nulled if the class is not set ?
```

Comment:
Looking at my local deployment of Epinio, `INGRESS_CLASS_NAME` is not set, neither is `--ingress-class-name`  and I see `null` in the values.yaml when I export an app. I guess we can remove the TODO.

Reply:
done: ee24af8df5a42ded79da7556a09482872adc6980